crates/formats/src/param.rs

```diff
@@ -0,0 +1,594 @@
+use core::str;
+use std::{borrow::Cow, ffi::CStr, marker::PhantomData};
+
+use thiserror::Error;
+use zerocopy::{FromBytes, FromZeroes, Unaligned, BE, LE, U16, U32, U64};
+
+/// Traits used to represent the varying endiannes, offset size and string encoding
+/// used it param files at compile time.
+pub mod traits {
+    use std::{borrow::Cow, ffi::CStr};
+
+    use utf16string::WStr;
+    use zerocopy::{ByteOrder, FromBytes, Unaligned, U16};
+
+    pub trait StaticBO: 'static + ByteOrder {}
+    impl<BO: 'static + ByteOrder> StaticBO for BO {}
```

Comment:
Is this just so you can avoid typing 'static on the traits below?

Reply:
Yeah. This bound comes from `utf16string`'s `WStr`: 
```rs
pub struct WStr<E: 'static + ByteOrder> {
    _endian: PhantomData<&'static E>,
    raw: [u8],
}
```
Not sure why the author used `&'static E` here when `fn() -> E` would have avoided the lifetime bound on `E`.

Would you prefer that we explicitly type `'static + ByteOrder` instead?